### PR TITLE
[LLVM][AArch64] Add assembly/disassembly of SVE BFSCALE instruction

### PR DESCRIPTION
**Pull request by Lukacma**
This patch add assembly/disassembly and tests for sve bfscale instruction according to https://developer.arm.com/documentation/ddi0602 .

### CODE REVIEW

**Review by jthackray — APPROVED** [review]

File: llvm/lib/Target/AArch64/AArch64SVEInstrInfo.td [hunk 1]
```diff
@@ -4163,7 +4163,9 @@ defm BFMIN_ZPZZ   : sve_fp_2op_p_zds_zeroing_bfloat<int_aarch64_sve_fmin>;
 defm BFMAX_ZPZZ   : sve_fp_2op_p_zds_zeroing_bfloat<int_aarch64_sve_fmax>;
 } // HasSVEB16B16, UseExperimentalZeroingPseudos
 
-
+let Predicates = [HasSVEBFSCALE] in {
+  def BFSCALE_ZPZZ : sve_fp_2op_p_zds_bfscale<0b1001, "bfscale", DestructiveBinaryComm>;
```
Comment (SpencerAbson) [inline llvm/lib/Target/AArch64/AArch64SVEInstrInfo.td:4167]: I'm not sure `DestructiveBinaryComm` is correct for `bfscale`, I think it should use `DestructiveBinary`.

File: llvm/lib/Target/AArch64/SVEInstrFormats.td [hunk 2]
```diff
@@ -2174,6 +2174,10 @@ multiclass sve_fp_2op_p_zds_bfloat<bits<4> opc, string asm, string Ps,
   def : SVE_3_Op_Pat<nxv8bf16, op, nxv8i1, nxv8bf16, nxv8bf16, !cast<Instruction>(NAME)>;
 }
 
+class  sve_fp_2op_p_zds_bfscale<bits<4> opc, string asm,  DestructiveInstTypeEnum flags> : sve_fp_2op_p_zds<0b00, opc, asm, ZPR16>{
```
Comment (CarolineConcatto) [inline llvm/lib/Target/AArch64/SVEInstrFormats.td:2177]: nit: formating

class  sve_fp_2op_p_zds_bfscale<bits<4> opc, string asm,  DestructiveInstTypeEnum flags> 
  : sve_fp_2op_p_zds<0b00, opc, asm, ZPR16>{
  let DestructiveInstType = flags;
  }

File: llvm/lib/Target/AArch64/AArch64SVEInstrInfo.td (hunk 1 above)
Comment (Lukacma) [reply llvm/lib/Target/AArch64/AArch64SVEInstrInfo.td]: Done

File: llvm/test/MC/AArch64/SVE2/bfscale-diagnostics.s [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+
+bfscale z31.d, p7/m, z31.h, z31.h
+// CHECK: [[@LINE-1]]:{{[0-9]+}}: error: invalid element width
+// CHECK-NEXT: bfscale z31.d, p7/m, z31.h, z31.h
+// CHECK-NOT: [[@LINE-1]]:{{[0-9]+}}:
+
+// ------------------------------------------------------------------------- //
+// Predicate register out of range
+
+bfscale z31.h, p8/m, z31.h, z31.h
```
[the remark this answers is not in the document]
Comment (SpencerAbson) [reply llvm/test/MC/AArch64/SVE2/bfscale-diagnostics.s:25]: Apologies - I should have specified that I meant a negative test, as using a zeroing predicate with `bfscale` should raise an error.

[the remark this answers is not in the document]
Comment (Lukacma) [reply llvm/test/MC/AArch64/SVE2/bfscale-diagnostics.s:25]: Done

**Review by SpencerAbson — APPROVED** [review]
Thanks, LGTM.

**Pull request merged**